### ISSUE TITLE
DbDependency using QueryBuilder

### ISSUE THREAD

**Issue by nadar** [yiisoft/yii2#13278]
### What steps will reproduce the problem?

When working with different sql schema the db dependency should be able to use the query builder to generate the sql statements.

```php
new DbDependency(['sql' => 'SELECT MAX(id) FROM admin_storage_folder WHERE is_deleted=0']);
```

The above example will work in mysql but not pgsql.

### Additional info

Our CMS system provides auto of the box caching mechanism, but db depencies are direct sql statements. It should be possible to provide query build with the query builder in order to provide independent sql schemas.

https://github.com/yiisoft/yii2/blob/master/framework/caching/DbDependency.php#L60

If there is another solution its not yet document in the guide nor api, let me know and will send a PR.  

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.10
| PHP version      | 7.x
| Operating system | Ubuntu

**Comment by samdark** [comment]
Can be done via http://chris-backhouse.com/Yii2-Output-the-SQL-from-a-query-builder/1027

**Comment by samdark** [comment]
Query support is a good idea though.

**Comment by nadar** [comment]
But its not possible to use the query builders [max()](http://www.yiiframework.com/doc-2.0/yii-db-query.html#max()-detail) method (or any other aggregation methods) as it will directly return the response value and its not possible to access the rawSql.

**Comment by klimov-paul** [comment]
Solution proposed at #13384

**Comment by nadar** [comment]
well done!

**Comment by klimov-paul** [comment]
Resolved by commit 95b7ffc066c03f2510ef4e6681e1ddf1494e916f
